packages/compass-serverstats/src/stores/current-op-store.js

```diff
@@ -79,79 +79,86 @@ const CurrentOpStore = Reflux.createStore({
     this.trigger(visErrors[this.overlayIndex], data);
   },
 
-  currentOp: function() {
-    if (this.dataService) {
-      this.dataService.currentOp(false, (error, response) => {
-        // Trigger error banner changes
-        if (error === null && this.errored.length > 0 && this.errored[this.errored.length - 1] !== null) { // Trigger error removal
-          Actions.dbError({'op': 'currentOp', 'error': null });
-        } else if (error !== null) {
-          Actions.dbError({'op': 'currentOp', 'error': error });
-        }
-        this.errored.push(error);
+  currentOp: async function() {
+    if (!this.dataService) {
+      return;
+    }
 
-        // Update op list if error
-        if (error !== null || this.starting) {
-          this.allOps.push([]);
-        }
+    let error = null; let response;
+
+    try {
+      response = this.dataService.currentOp(false);
+    } catch (err) {
+      error = err;
+    }
```

Comment:
There is a bunch of very convoluted logic down the code flow in this and similar actions that uses error in state in combination with error returned by `dataService` so by refactoring it that way I am just avoiding the need to untangle this state usage

Reply:
To understand the `compass-serverstats` package's store you had to look at all of the stores at once 😭 😆

Reply:
Yeah... I totally wasn't ready for this 😓